apps/mobile-wallet/src/screens/Settings/SettingsScreen.tsx

```diff
@@ -308,16 +262,46 @@ const SettingsScreen = ({ navigation, ...props }: ScreenProps) => {
                   variant="accent"
                   onPress={() => {
                     props.onClose && props.onClose()
-                    setAuthCallback(() => () => setIsMnemonicModalVisible(true))
-                    setAuthenticationPrompt("Verify it's you")
-                    setLoadingText('Verifying...')
-                    setIsAuthenticationModalOpen(true)
+
+                    if (biometricsRequiredForAppAccess) {
+                      triggerBiometricsPrompt({
+                        successCallback: () => setIsMnemonicModalVisible(true)
+                      })
+                    } else {
+                      setIsMnemonicModalVisible(true)
+                    }
                   }}
                 />
               </ScreenSection>
             </ModalContent>
           )}
         />
+      </Portal>
+
+      <Portal>
+        <BottomModal
+          isOpen={isSwitchNetworkModalOpen}
+          onClose={() => setIsSwitchNetworkModalOpen(false)}
+          Content={(props) => (
+            <SwitchNetworkModal
+              onClose={() => setIsSwitchNetworkModalOpen(false)}
+              onCustomNetworkPress={() => navigation.navigate('CustomNetworkScreen')}
+              {...props}
+            />
+          )}
+        />
+
+        <BottomModal
+          isOpen={isCurrencySelectModalOpen}
+          onClose={() => setIsCurrencySelectModalOpen(false)}
+          Content={(props) => <CurrencySelectModal onClose={() => setIsCurrencySelectModalOpen(false)} {...props} />}
+        />
+
+        <BottomModal
+          isOpen={isMnemonicModalVisible}
+          onClose={() => setIsMnemonicModalVisible(false)}
+          Content={(props) => <MnemonicModal {...props} />}
+        />
```

Comment:
This is just the same code, but moved, to fix a glitch when opening a modal right after closing another one.